app/controllers/users/noticed_notifications_controller.rb

```diff
@@ -12,6 +12,7 @@ def mark_as_read
     notification = NoticedNotification.find(params[:notification_id])
     notification.update(read_at: Time.zone.now)
     project = notification.params[:project]
+    # assignment = notification.params[:assignment]
     redirect_to user_project_path(project.author, project)
```

Comment:
```suggestion
    if notification.type == "NewAssignmentNotification"
      assignment = notification.params[:assignment]
      group = assignment.group
      redirect_to group_assignment_path(group, assignment)
    else
      project = notification.params[:project]
      redirect_to user_project_path(project.author, project)
    end
```

This is for redirection purpose, when user clicks to notification where it should get redirected, previously we had star and fork notification for which user should get redirected to project path, but for assignment it should get redirect to assignment path which can be achieved using this.

Test it!